### ISSUE TITLE
Fix S1144 FN: Unused private methods with attributes

### ISSUE THREAD

**Issue by antonioaversa** [SonarSource/sonar-dotnet#8348]
The presence of any attribute on a `private` method makes the S1144 rule not report:
```cs
using Microsoft.JSInterop;

public class S6798_CSharpOnly
{
    private void JSInvokablePrivateMethodInPartialRazorComponent() { } // Noncompliant: this is detected
    [JSInvokable] void JSInvokableImplicitelyPrivateMethodInPartialRazorComponent() { } // FN: this is not
}
```

There doesn't seem to be a valid reason for this difference in behavior.

Reproduced by #8349

**Comment by Tim-Pohlmann** [comment]
@antonioaversa RSPEC lists `members with attributes` as an exception. It is unclear to me whether you missed that or you are challenging the exception.
